src/components/smart/OeZoneerderZonePanel.vue

```diff
@@ -0,0 +1,444 @@
+
+onMounted(() => {
+  emit('zone-panel:mounted', elementRef.value);
+});
+
+onUnmounted(() => {
+  map
+    .getControls()
+    .getArray()
+    .filter((control) => control['element'].classList.contains('zone-panel'))
+    .forEach((control) => map.removeControl(control));
+});
+
+const perceelSelectCallback = (evt: MapBrowserEvent<UIEvent>) => {
+  crabService.searchPerceel(evt.coordinate, mapProjection.getCode(), agivGrbUrl).then((result) => {
+    geoJsonFormatter.readFeatures(result).forEach((perceel) => {
+      drawPerceel(perceel);
+    });
+  });
+};
+
+const drawInteractions: { [type: string]: Draw } = _createInteractions();
+
+function togglePanel() {
+  panelOpen.value = !panelOpen.value;
+}
+
+function updateWKTString(value: string) {
+  WKTString.value = value;
+}
+
+function _createVectorLayer(options: { color: ColorLike; fill: ColorLike; title: string }) {
+  const getText = (feature: FeatureLike) =>
+    new OlText({
+      font: '10px Verdana',
+      text: feature.get('name') || '',
+      fill: new Fill({ color: options.color }),
+      stroke: new Stroke({ color: '#fff', width: 3 }),
+    });
+
+  const getStyle = (feature: FeatureLike) =>
+    new Style({
+      stroke: new Stroke({ color: options.color, width: 3 }),
+      fill: new Fill({ color: options.fill }),
+      text: getText(feature),
+    });
+  const vLayer = new VectorLayer({
+    source: new VectorSource(),
+    style: getStyle,
+    visible: true,
+  });
+  vLayer.set('title', options.title);
+  vLayer.set('type', 'overlay');
+
+  return vLayer;
+}
+
+function toggleDrawZone(drawZoneEnabled = false, type: IDrawGeomType = 'Polygon') {
+  resetSelect();
+  addingWKT.value = false;
+  map.getInteractions().pop();
+  activeDrawType.value = drawZoneEnabled ? type : undefined;
+  for (const [drawType, interaction] of Object.entries(drawInteractions)) {
+    if (drawType === type) {
+      interaction.setActive(drawZoneEnabled);
+      map.addInteraction(interaction);
+    } else {
+      interaction.setActive(false);
+    }
+  }
+}
+
+function resetSelect() {
+  selectPerceel.value = false;
+  if (perceelSelectCallback) {
+    map.removeEventListener('click', perceelSelectCallback as Listener);
+  }
+}
+
+function drawWKTZone() {
+  const wktParser = new WKT();
+  try {
+    const featureFromWKT = wktParser.readFeature(WKTString.value);
+    const name = `Polygoon ${polygonIndex++}`;
+    featureFromWKT.setProperties({
+      name: name,
+    });
+    drawLayer.getSource()?.addFeature(featureFromWKT);
+    geometryObjectList.value.push(name);
+    zoomToFeatures();
+    WKTString.value = '';
+  } catch (error) {
+    console.error(error, 'Dit is een ongeldige WKT geometrie.');
+  }
+}
+
+function zoomToFeatures() {
+  const extent = drawLayer.getSource()?.getExtent();
+  if (!extent) return;
+
+  zoomToExtent(extent);
+}
+
+function showWktInput() {
+  toggleDrawZone(false);
+  addingWKT.value = true;
+}
+
+function startPerceelSelect() {
+  toggleDrawZone(false);
```

Comment:
eventueel hier de cursor wijzigen naar pointer ofzo, zodat duidelijk is dat er een functie geselecteerd is?